crates/wash-lib/src/cli/mod.rs

```diff
@@ -223,14 +219,11 @@ impl TryFrom<CliConnectionOpts> for WashConnectionOptions {
         }: CliConnectionOpts,
     ) -> Result<WashConnectionOptions> {
         // Attempt to load a context, falling back on the default if not supplied
-        let ctx = if let Some(context) = context {
-            Some(load_context(context)?)
-        } else if let Ok(ctx_dir) = context_dir(None) {
-            let ctx_dir = ContextDir::new(ctx_dir)?;
-            ensure_host_config_context(&ctx_dir)?;
-            Some(ctx_dir.load_default_context()?)
+        let ctx_dir = ContextDir::new()?;
+        let ctx = if let Some(context_name) = context {
+            Some(ctx_dir.load_context(&context_name)?)
         } else {
-            None
+            ctx_dir.load_default_context().ok()
```

Comment:
NIT: `ContextDir::load_name_or_default` ? `ContextDir::load_named`? `ContextDir::load_by_name` ?

Reply:
I think I prefer keeping `load_context` and `load_default_context` separate